app/questionnaire/placeholder_parser.py

```diff
@@ -96,23 +81,41 @@ def _parse_transforms(self, transform_list: Sequence[Mapping]):
         transformed_value = None
 
         for transform in transform_list:
-            transform_args: Dict[str, Union[None, str, List[str]]] = {}
+            transform_args: Dict[
+                str, Union[None, str, List[Union[str, int, float]]]
```

Comment:
I think answers can either be `str`, `int`, `Decimal`, `dict` (duration and address answers), or `list[str]` (checkbox answers).
```suggestion
                str, Union[None, str, int, Decimal, dict, list[str]]
```

We should also be using the built in types rather than the ones from `typing` e.g. `list` rather than `List`. I can see that this matches what was already there, but as long as it doesn't cause any problems we should update it.